File: native/swift/swift-export-standalone/testData/execution/innerClass/innerClass.kt
// KIND: STANDALONE
// MODULE: InnerClass
// FILE: main.kt
class Outer {
    inner class Inner {
        inner class InnerInner
    }
}

class OuterWithParam(val outerParam: Int) {
    inner class InnerWithParam(val innerParamA: Int, val innerParamB: Int) {
        inner class InnerInnerWithParam(val innerInnerParam: Int) {
            fun getOuter(): Int = outerParam
            fun getInnerA(): Int = innerParamA
            fun getInnerB(): Int = innerParamB
            fun getInnerInner(): Int = innerInnerParam
        }
    }
}

File: native/swift/swift-export-standalone-integration-tests/simple/testData/execution/innerClass/innerClass.swift
import InnerClass
import Testing

@Test
func testInnerClass() {
    let outer = Outer()
    let inner = Outer.Inner(outer__: outer)
    let innerInner = Outer.Inner.InnerInner(outer__: inner)
}

@Test
func testNestedInnerClassWithParameters() throws {
    let outer = OuterWithParam(outerParam: 0)
    let inner = OuterWithParam.InnerWithParam(innerParamA: 1, innerParamB: 2, outer__: outer)
    let innerInner = OuterWithParam.InnerWithParam.InnerInnerWithParam(innerInnerParam: 3, outer__: inner)
    try #require(innerInner.getOuter() == 0)
    try #require(innerInner.getInnerA() == 1)
    try #require(innerInner.getInnerB() == 2)
    try #require(innerInner.getInnerInner() == 3)
}